freqtrade/rpc/api_server/ws/proxy.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Tuple, Union
+
+from fastapi import WebSocket as FastAPIWebSocket
+from websockets import WebSocketClientProtocol as WebSocket
+
+from freqtrade.rpc.api_server.ws.types import WebSocketType
+
+
+class WebSocketProxy:
```

Comment:
So this was for the WebSocketChannel to have a unified interface on both of the websocket objects. Personally I think either way would work. If we went the abstract parent class route, we'd have the interface definition, and 2 subclasses. The WebSocketChannel would then have to take care initializing which one depending on maybe a parameter we can pass to it's init func. Realistically though, I'd say both ways are valid. If we were to maybe extend it with more than 2 types then it would become a problem as that class would blow up very fast.

Reply:
well currently it's problematic for another reason - running `mypy freqtrade` throws quite some errors in that class - and i'm not sure how they can be addressed without `#type: ignore` - which we really shouldn't do - as we KNOW the type we use.